tests/test.linter.js

```diff
@@ -183,6 +183,21 @@ describe('Linter', function() {
       });
   });
 
+  it('JSScanner should ignore eslint ignore patterns and .eslintrc', () => {
+    var addonLinter = new Linter({_: [
+      'tests/fixtures/webextension_node_modules_bower']});
+
+    // Stub print to prevent output.
+    addonLinter.print = sinon.stub();
+
+    assert.equal(addonLinter.collector.warnings.length, 0);
+
+    return addonLinter.scan()
+      .then(() => {
+        assert.equal(addonLinter.collector.warnings.length, 0);
```

Comment:
I added some comments about what it's proving in 78e6c41

If I remove the new `patterns` option the test fails with exactly 2 warnings.